Detectors/Align/src/AliAlgDetITS.cxx

```diff
@@ -59,82 +63,59 @@ void AliAlgDetITS::DefineVolumes()
 {
   // define ITS volumes
   //
-  const int kNSPDSect = 10;
-  AliAlgVol *volITS = 0, *hstave = 0, *ladd = 0;
-  AliAlgSensITS* sens = 0;
-  //
-  int labDet = GetDetLabel();
-  AddVolume(volITS = new AliAlgVol("ITS", labDet));
-  //
-  // SPD
-  AliAlgVol* sect[kNSPDSect] = {0};
-  for (int isc = 0; isc < kNSPDSect; isc++) { // sectors
-    int iid = labDet + (10 + isc) * 10000;
-    AddVolume(sect[isc] = new AliAlgVol(Form("ITS/SPD0/Sector%d", isc), iid));
-    sect[isc]->SetParent(volITS);
+  int labNS = GetDetLabel() + mMapping.getNChips(); // non-sensors labels will start from this
+  auto volITS = AddVolume(new AliAlgVol(GeometryTGeo::composeSymNameITS(), GetDetLabel()));
+  using MP = o2::itsmft::ChipMappingITS;
+
+  auto addVolumeChip = [this](int ilr, int ist, int isst, int imd, int ich, AliAlgVol* parent) {
+    auto volChip = AddVolume(new AliAlgSensITS(GeometryTGeo::composeSymNameChip(ilr, ist, isst, imd, ich), ilr, this->fSensors.GetEntriesFast()));
+    volChip->SetParent(parent);
+  };
+
+  auto addVolumeModule = [&, this](int ilr, int ist, int isst, int imd, AliAlgVol* parent) {
+    auto lrType = MP::layer2RUType(ilr);
+    if (imd >= 0) {
+      auto volMod = AddVolume(new AliAlgVol(GeometryTGeo::composeSymNameModule(ilr, ist, isst, imd), labNS++));
+      volMod->SetParent(parent);
+      parent = volMod;
+    }
+    for (int ich = 0; ich < MP::getNChipsPerModuleSB(lrType); ich++) {
+      addVolumeChip(ilr, ist, isst, imd, ich, parent);
+    }
+  };
+
+  auto addVolumeSStave = [&, this](int ilr, int ist, int isst, AliAlgVol* parent) {
```

Comment:
Typo?

Reply:
where?

Reply:
addVolumeSStave = addVolumeStave or addVolumesStave ?

Reply:
addVolumeS[ub]Stave: the staves of outer 4 layers are made of 2 sub-staves